src/main/java/az/edu/turing/business/FamilyService.java

```diff
@@ -96,7 +99,11 @@ public Family getFamilyById(int index) {
 
     public Set<Pet> getPets(int familyIndex) {
         Family family = getFamilyById(familyIndex);
-        return family.getPets();
+        if (family != null) {
+            Set<Pet> pets = family.getPets();
+            return pets != null ? new HashSet<>(pets) : new HashSet<>();
```

Comment:
I think no need checking pets' nullability here instead you can initialize pet default in Family class itself